### ISSUE TITLE
Doctrine\Common\Persistence\ObjectManager used in Faker\ORM\Doctrine\Populator was moved in another namespace

### ISSUE THREAD

**Issue by gnutix** [FakerPHP/Faker#318]
### Summary

The class `Doctrine\Common\Persistence\ObjectManager` was moved to `Doctrine\Persistence\ObjectManager` in Doctrine 2.8.x. It is used in `Faker\ORM\Doctrine\Populator`.

### Versions

|                | Version |
|:---------------|:--------|
| PHP            | 8.0.3   |
| `fakerphp/faker`  | 1.14.1   |

### Expected

The code bridging Faker and Doctrine should probably be extracted to a separate package `fakerphp/faker-doctrine` in order to allow proper dependencies management using Composer.

**Comment by pimjansen** [comment]
This is correct. To keep the BC change we should test if the method exists and call the correct method from there instead. Can you raise a Pr for this?

Keep in mind that the ORM part is not part of the Faker core and therefor also be dropped from 2.x forward. It will be there as a separate third party extension though. But due the fact ORM integration rests more on Doctrine and not on Faker here it will be better to maintain from that angle.

**Comment by gnutix** [comment]
It's not a method change, but a class that moved from one namespace to another. I don't know how a public package is supposed to deal with such cases of "supporting multiple versions of a 3rd party library". And I guess that, as we both said, splitting Faker core with bridge code to separate repositories is a good way to go... If you plan to do that for a next major release, that seems fine to me.

In the meantime, I've fixed the namespaces by applying a Composer patch when installing fakerphp/faker in my project.

**Comment by pimjansen** [comment]
Indeed. If we want a fix for the current implementation we should check if the class exists to support both i guess
